File: assignments/a4/TYPE1/Java/PrimTypeExp.java
//PrimTypeExp:top//
//PrimTypeExp:import//
import java.util.*;

// <typeExp>:PrimTypeExp ::= <primType>
public class PrimTypeExp extends TypeExp /*PrimTypeExp:class*/ {

    public static final String $className = "PrimTypeExp";
    public static final String $ruleString =
        "<typeExp>:PrimTypeExp ::= <primType>";

    public PrimType primType;

    public PrimTypeExp(PrimType primType) {
//PrimTypeExp:init//
        this.primType = primType;
    }

    public static PrimTypeExp parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<typeExp>:PrimTypeExp", scn$.lno);
        PrimType primType = PrimType.parse(scn$, trace$);
        return new PrimTypeExp(primType);
    }

    public Type toType() {
      return primType.toType();
    }

//PrimTypeExp//
}

File: assignments/a4/TYPE1/Python/PrimTypeExp.py
#PrimTypeExp:top#
#PrimTypeExp:import#

# <typeExp>:PrimTypeExp ::= <primType>
class PrimTypeExp(TypeExp): #PrimTypeExp:class#

    className = "PrimTypeExp"
    ruleString = "<typeExp>:PrimTypeExp ::= <primType>"
    primType = None

    def __init__(primType):
        #PrimTypeExp:init#
        self.primType = primType

#PrimTypeExp#
